### PR TITLE
Update CI runners

### PR DESCRIPTION
**Pull request by albinahlback**
In particular, remove x86 macOS runner since they are deprecated on Github.

### CODE REVIEW

File: .github/workflows/CI.yml
Comment (albinahlback) [inline .github/workflows/CI.yml:1]: @edgarcosta How can I find Clang systematically here? I believe Clang is not on the path, so I have to manually get the path, but I don't want to guess the version of `llvm` (see line 219).

Comment (edgarcosta) [reply .github/workflows/CI.yml:1]: I got the runner working with the latest llvm, but then `make check` fails

Comment (albinahlback) [reply .github/workflows/CI.yml:1]: Yeah, newer LLVM doesn't seem to like FLINT.

**Comment by albinahlback** [comment]
@fredrik-johansson I think I have fixed a lot of errors for generic macros (like `add_s{n}a{2n}`).  We cannot have something like
```c
# define add_ssaaaa(s1, s0, a1, a0, b1, b0) \
  do { \
    ulong __t0 = (a0); \
    (s0) = (a0) + (b0); \
    (s1) = (a1) + (b1) + ((ulong) (s0) < __t0); \
  } while (0)

# define add_sssaaaaaa(s2, s1, s0, a2, a1, a0, b2, b1, b0) \
  do { \
    ulong __t0, __t1; \
    add_ssaaaa(__t0, s0, (ulong) 0, a0, (ulong) 0, b0); \
    add_ssaaaa(__t1, s1, (ulong) 0, a1, (ulong) 0, b1); \
    add_ssaaaa(s2, s1, (a2) + (b2), s1, __t1, __t0); \
  } while (0)
```
since `__t0` is used in both `add_sssaaaaaa` and `add_ssaaaa`. Not sure why it causes a problem, though, just that this fixes the problem.  `src/test`, `mpn_extras` and `ulong_extras` seems to work fine now, but one matrix multiplication in `nfloat` seems to fail; I'm unable to track where this error stems from.  Temporarily, we can check the fails on generic macros by commenting out everything between line 28 and line 58 in `longlong.h`, but we should integrate this into our CI.

**Comment by fredrik-johansson** [comment]
How should I ``configure`` FLINT to reproduce the ``nfloat`` failure?

**Comment by albinahlback** [comment]
> How should I `configure` FLINT to reproduce the `nfloat` failure?

You can configure as normal, but just comment out every `longlong_*.h` inclusion inside `longlong.h` (so only generic macros are being used).

**Comment by fredrik-johansson** [comment]
Found it: ``FLINT_MPN_MUL_2X2H`` in ``nfloat/dot.c`` needs unique variable names too.

**Comment by edgarcosta** [comment]
Well done!

**Comment by fredrik-johansson** [comment]
Best I guess would be to choose very unique names for the temporaries in ``add_ssaaaa`` et al to minimize the risk of downstream breakage.

**Comment by albinahlback** [comment]
I will merge this when all GCC tests passes. Clang seems to be really buggy -- we have two different bugs popping up.  For Ubuntu Clang, it doesn't want to compile a test file, and for Apple M1 it wrongly assumes that `__builtin_clzl` always returns something non-zero, which results in the failure of `sdiv_qrnnd`.

**Pull request merged**